pandas/util/testing.py

```diff
@@ -878,8 +879,13 @@ def _get_ilevel_values(index, level):
             # get_level_values may change dtype
             _check_types(left.levels[level], right.levels[level], obj=obj)
 
-    # skip exact index checking when `check_categorical` is False
-    if check_exact and check_categorical:
+    if isinstance(left, DatetimeIndex):
+        # by now we know right is also a DatetimeIndex
+        assert_numpy_array_equal(left.asi8, right.asi8)
+        assert tz_compare(left.tz, right.tz)
```

Comment:
why is any of these changes needed? this is really suspect. instead the tz comparison should be done in assert_index_equal (which IIRC) is already done, no?

Reply:
This is very kludgy and all needs to be pushed up the stack.  I tentatively think this particular check will be fixed by making `Datetime64TZDtype.__eq__` use tz_compare, at which point the changes here won't be necessary.